infra/templates/azure-ase-app-service-single-region/docs/design.md

```diff
@@ -0,0 +1,161 @@
+# Deploying App Service into ASE 
+
+v0.1 - 6/19/2019 
+
+## Introduction 
+
+Some of Cobalt’s enterprise customers have strict security guidelines that have led them to leverage Azure App Service Environments (ASE) as their primary infrastructure target when deploying to Azure. This is a natural choice as ASE enables customers to deploy to a single-tenant fleet of hosts running in a secure and isolated network that can be hooked into internal, client-owned networks without traversing the public internet. 
+
+This document outlines how Cobalt can be extended to meet the use cases of these customers. The intended audience of this document is the development and product teams working on Cobalt and related projects. 
+
+## In Scope 
+
+- Identify deployment topology needed by the customer 
+- Identify key Terraform templates needed for deployment
+- Identify key Terraform modules needed for deployment
+- Identify gaps in Terraform provider templates
+
+## Out of scope 
+
+- External customer sign-off
```

Comment:
Mostly due to the fact that it will take > 1 sprint to write the design and get customer sign-off. I also think it will be difficult to get sign-off in isolation of the other pending designs such as CI/CD, security (authn/z), template breakdown, etc...

Thoughts?